src/mdns/mdns_avahi.cpp

```diff
@@ -413,6 +424,51 @@ void PublisherAvahi::CallHostOrServiceCallback(AvahiEntryGroup *aGroup, otbrErro
     }
 }
 
+otbrError PublisherAvahi::RetryPublishService(const Services::iterator &aServiceIt)
```

Comment:
This method contains duplicated logic of PublishService. 

Can we reuse the same logic?

Reply:
I tried doing so initially but I found it hard to achieve it, because there are some logic about updating the service in `PublishService`will result in a wrong behavior when retrying publishing the service.

Reply:
SGTM if it's hard to do.